cpp/src/parquet/encryption/internal_file_encryptor.cc

```diff
@@ -52,8 +52,15 @@ InternalFileEncryptor::InternalFileEncryptor(FileEncryptionProperties* propertie
 void InternalFileEncryptor::WipeOutEncryptionKeys() {
   properties_->WipeOutEncryptionKeys();
 
-  for (auto const& i : all_encryptors_) {
-    i->WipeOut();
+  for (auto const& i : meta_encryptor_) {
+    if (i != nullptr) {
```

Comment:
If we ensure that `AesEncryptor::Make` and `AesDecryptor::Make` always return a non-null value, we don't need those checks.

Reply:
```
  // Key must be 16, 24 or 32 bytes in length. Thus there could be up to three
  // types of meta_encryptors and data_encryptors.
  std::unique_ptr<encryption::AesEncryptor> meta_encryptor_[3];
  std::unique_ptr<encryption::AesEncryptor> data_encryptor_[3];
```

This is as said before, since it might only has one?

Reply:
Ah, sorry, I had forgotten about this. Nevermind.